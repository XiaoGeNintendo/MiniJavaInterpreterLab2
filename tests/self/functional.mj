//Note this is not required by the lab requirement
class Int{
    int count = 0;

    Int(){
        this(0);
    }

    Int(int no){
        count=no;
    }
    Int set(int i) {
        count = i;
        return this;
    }
    Int add(int i) {
        count += i;
        return this;
    }
    Int doubles(){
        count *= 2;
        return this;
    }

    Int print(){
        println(count);
        return this;
    }
}

int main(){
    Int a=new Int();
    Int b=new Int(3);
    a.print().set(1).print().add(2).print().doubles().print();
    b.print().set(2).print().add(3).print().doubles().print();
    return 0;
}