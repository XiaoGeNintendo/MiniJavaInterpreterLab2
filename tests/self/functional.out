0
1
3
6
3
2
5
10
Process exits with the code 0.